pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -165,9 +146,13 @@ public SuccessResponse addTable(String tableConfigStr, @Context HttpHeaders http
       throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.BAD_REQUEST, e);
     }
     try {
-      ensureMinReplicas(tableConfig);
-      ensureStorageQuotaConstraints(tableConfig);
-      verifyTableConfigs(tableConfig);
+      try {
+        TableConfigUtils.ensureMinReplicas(tableConfig, _controllerConf.getDefaultTableMinReplicas());
+        TableConfigUtils.ensureStorageQuotaConstraints(tableConfig, _controllerConf.getDimTableMaxSize());
+        checkHybridTableConfig(TableNameBuilder.extractRawTableName(tableName), tableConfig);
+      } catch (Exception e) {
+        throw new PinotHelixResourceManager.InvalidTableConfigException(e);
```

Comment:
Should we consider taking `InvalidTableConfigException` out of `PinotHelixResourceManager`?

Reply:
Done. Moved it to a package along with `TableAlreadyExistsException` and `ControllerApplicationException`